6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1

```diff
@@ -205,231 +199,342 @@ Function CreateAppRole([string] $types, [string] $name, [string] $description)
-                                                  -HomePage "http://localhost:4200" `
-                                                  -ReplyUrls "http://localhost:4200" `
-                                                  -IdentifierUris "https://$tenantName/ProfileSPA" `
-                                                  -PublicClient $False
-
-   # create the service principal of the newly created application 
-   $currentAppId = $clientAadApplication.AppId
-   $clientServicePrincipal = New-AzureADServicePrincipal -AppId $currentAppId -Tags {WindowsAzureActiveDirectoryIntegratedApp}
-
-   # add the user running the script as an app owner if needed
-   $owner = Get-AzureADApplicationOwner -ObjectId $clientAadApplication.ObjectId
-   if ($owner -eq $null)
-   { 
-        Add-AzureADApplicationOwner -ObjectId $clientAadApplication.ObjectId -RefObjectId $user.ObjectId
-        Write-Host "'$($user.UserPrincipalName)' added as an application owner to app '$($clientServicePrincipal.DisplayName)'"
-   }
+    # Add Required Resources Access (from 'client' to 'service')
+    Write-Host "Getting access from 'client' to 'service'"
+    $requiredPermission = GetRequiredPermissions -applicationDisplayName "ProfileAPI"`
+        -requiredDelegatedPermissions "access_as_user"
 
+    $requiredResourcesAccess.Add($requiredPermission)
+    Write-Host "Added 'service' to the RRA list."
+    # Useful for RRA additions troubleshooting
+    # $requiredResourcesAccess.Count
+    # $requiredResourcesAccess
+    
+    Update-MgApplication -ApplicationId $currentAppObjectId -RequiredResourceAccess $requiredResourcesAccess
+    Write-Host "Granted permissions."
+    
 
-   Write-Host "Done creating the client application (ProfileSPA)"
+    # print the registered app portal URL for any further navigation
+    Write-Host "Successfully registered and configured that app registration for 'ProfileSPA' at `n $clientPortalUrl" -ForegroundColor Green 
 
-   # URL of the AAD application in the Azure portal
-   # Future? $clientPortalUrl = "https://portal.azure.com/#@"+$tenantName+"/blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
-   $clientPortalUrl = "https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/CallAnAPI/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
-   Add-Content -Value "<tr><td>client</td><td>$currentAppId</td><td><a href='$clientPortalUrl'>ProfileSPA</a></td></tr>" -Path createdApps.html
+    # Configure known client applications for service 
+    Write-Host "Configure known client applications for the 'service'"
+    $knowApplications = New-Object System.Collections.Generic.List[System.String]
+    $knowApplications.Add($clientAadApplication.AppId)
+    Update-MgApplication -ApplicationId $currentAppObjectId -Api @{KnownClientApplications = $knowApplications}
```

Comment:
Written as a reminder that we need to update the `$currentAppObjectId` to be pointed to the API project. Right now this is sending the SPA as a known client to the SPA application in Azure.

Reply:
done!